docs/tutorials/notes-mvvm/includes/viewmodel-note.md

```diff
@@ -0,0 +1,130 @@
+    using System.Windows.Input;
+    
+    namespace Notes.ViewModels;
+    
+    internal class NoteViewModel : ObservableObject, IQueryAttributable
+    {
+        private Models.Note _note;
+
+    }
+    ```
+
+    This code is the blank `Note` viewmodel where you'll add properties and commands to support the `Note` view. Notice that the `CommunityToolkit.Mvvm.ComponentModel` namespace is being imported. This namespace provides the `ObservableObject` used as the base class. You'll learn more about `ObservableObject` in the next step. The `CommunityToolkit.Mvvm.Input` namespace is also imported. This namespace provides some command-types that invoke methods asynchronously.
+
+    The `Models.Note` model is being stored as a private field. The properties and methods of this class will use this field.
+
+01. Add the following properties to the class:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="properties":::
+
+    The `Date` and `Identifier` properties are simple properties that just retrieve the corresponding values from the model.
+
+    > [!TIP]
+    > For properties, the `=>` syntax creates a get-only property where the statement to the right of `=>` must evaluate to a value to return.
+
+    The `Text` property first checks if the value being set is a different value. If the value is different, that value is passed on to the model's property, and the `OnPropertyChanged` method is called.
+
+    The `OnPropertyChanged` method is provided by the `ObservableObject` base class. This method uses the name of the calling code, in this case, the property name of **Text**, and raises the `ObservableObject.PropertyChanged` event. This event supplies the name of the property to any event subscribers. The binding system provided by .NET MAUI recognizes this event, and updates any related bindings in the UI. For the **note viewmodel**, when the `Text` property changes, the event is raised, and any UI element that is bound to the `Text` property is notified that the property changed.
+
+01. Add the following command-properties to the class, which are the commands that the view can bind to:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="commands":::
+
+01. Add the following constructors to the class:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="ctor":::
+
+    These two constructors are used to either create the viewmodel with a new backing model, which is an empty note, or to create a viewmodel that uses the specified model instance.
+
+    The constructors also setup the commands for the viewmodel. Next, add the code for these commands.
+
+01. Add the `Save` and `Load` methods:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="command_methods":::
```

Comment:
Wondering why these are public, when they are invoked only by commands?

Reply:
Just as a matter of habit, I always end up coding public and then restricting down when I don't want someone to access something. However, I should stop doing that :D I'll update these.